main/admin/settings.lib.php

```diff
@@ -638,6 +738,132 @@ function uploadStylesheet($values, $picture)
     return $result;
 }
 
+/**
+ * Creates the folder (if needed) and uploads the plugin in it. If the plugin
+ * is already there and the folder is writeable, overwrite.
+ *
+ * @param array $values  the values of the form
+ * @param array $file   the file passed to the upload form
+ * @param array $officialPlugins A list of official plugins that cannot be uploaded
+ *
+ * @return bool
+ */
+function uploadPlugin($values, $file, $officialPlugins)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $extraction_path is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5078559)